### ISSUE TITLE
Create more warehouse size options in "ELT" terraform module

### ISSUE THREAD

**Issue by ian-r-rose** [cagov/data-infrastructure#297]
We currently just have three Snowflake warehouse sizes, allowing for not-very-fine-grained tuning of tasks to the compute power. We should extend the terraform module to be a bit more configurable on this front.

**Comment by jkarpen** [comment]
This should be completed before we need to spin up an account for HFA

**Comment by ram-kishore-odi** [comment]
Following are the standard snowflake warehouse sizes that can be used.
Using the the sizes from list below does not require contacting Snowflake.
We can add these to the ELT module to give more flexibility to choose from.

Standard Sizes:

X-Small: Good for small tasks and experimenting.
Small: Suitable for single-user workloads and development.
Medium: Handles moderate concurrency and data volumes.
Large: Manages larger queries and higher concurrency.
X-Large: Powerful for demanding workloads and data-intensive operations.
2X-Large: Double the capacity of X-Large.
3X-Large: Triple the capacity of X-Large.
4X-Large: Quadruple the capacity of X-Large.

Sizes larger than 4x often require contacting Snowflake. I also feel that they are not relevant to projects that ODI normally works on.

@ian-r-rose Can you please comment on whether you are thinking about providing additional size options alone or if you are thinking about any other options ?

**Comment by ram-kishore-odi** [comment]
PR was approved and code changes were merged into main by Ian.
